### PR TITLE
cli: Fix migrate create not working on Windows

### LINKED ISSUE #238: migrate create with seq does not handle dot dir prefix
**Describe the Bug**
When using the `create` subcommand with the `seq` flag, the command fails if `dir` begins with a `.`.

**Steps to Reproduce**
1. Create first migration:
```bash
mkdir test
migrate create -ext sql -dir ./test -seq first
```

2. Create second migration:
```bash
migrate create -ext sql -dir ./test -seq second
```

3. See error:
```
error: strconv.Atoi: parsing "test/000001": invalid syntax
```

**Expected Behavior**
`migrate create -seq` should create a second migration with the proper sequence number when `dir` begins with a `.`.

**Migrate Version**
`v4.4.0`

**Loaded Source Drivers**
```
Source drivers: gitlab, go-bindata, godoc-vfs, gcs, file, s3, github
```

**Loaded Database Drivers**
```
Database drivers: cassandra, crdb-postgres, postgres, stub, clickhouse, cockroach, cockroachdb, mongodb, mysql, postgresql, redshift, spanner, sqlserver
```

**Go Version**
```
go version go1.12.5 darwin/amd64
```

**Stacktrace**
Pretty sure it's this line: https://github.com/golang-migrate/migrate/blob/master/internal/cli/commands.go#L31

**Additional context**
`migrate create -seq` works just fine when `dir` does *not* begin with `.`.

### PR DESCRIPTION
**Pull request by toqueteos**
This fixes #238 in Windows by not using hardcoded `/` when dealing with paths.
Other `GOOS`es shouldn't be affected.

I simplified the `cleanDir` helper function, there's no need to append a trailing slash to paths.

### CODE REVIEW

**Comment by toqueteos** [comment]
Builds fail due to `DOCKER_API_VERSION=1.39` but `Maximum supported API version is 1.38`

File: internal/cli/commands.go [hunk 1]
```diff
@@ -73,7 +70,7 @@ func createCmd(dir string, startTime time.Time, format string, name string, ext
 		if seqDigits <= 0 {
 			log.fatalErr(errors.New("Digits must be positive"))
 		}
-		matches, err := filepath.Glob(dir + "*" + ext)
+		matches, err := filepath.Glob(filepath.Join(dir, "*"+ext))
```
Comment (dhui) [inline internal/cli/commands.go]: Regardless the final solution, I think we should include this change.

**Comment by toqueteos** [comment]
> How did you test your changes?
> 
> I believe this change will break `nextSeq()` which expects the `dir` to have a trailing slash when calling `strings.TrimPrefix()`.

I ran the tests in my Windows machine and inside a Linux VM.

**Comment by toqueteos** [comment]
`nextSeq` worked just fine and multiple `migrate create` calls succeeded.

We can't be sure files are created properly in tests unless we add a virtual filesystem dependency. What do you think about it?

**Comment by dhui** [comment]
I pulled your branch, tested creating migrations using the `seq` option, and encountered the following issues:
1. Migrations are not created in the specified directory
1. Migrations do not properly increment the version number

> We can't be sure files are created properly in tests unless we add a virtual filesystem dependency. What do you think about it?

In this specific case, we don't need an integration test to test for correctness. The existing tests test that the methods adhere to a "spec" (or define the "spec") and the "specs" for the methods are designed to work together. e.g. the output for `cleanDir()` is expected to be passed to `nextSeq()`.
Unfortunately, the specs aren't clearly documented right now and are only checked/enforced in the test...

**Comment by toqueteos** [comment]
Addressed in #352

**Pull request closed without merging**